src/client/extensionActivation.ts

```diff
@@ -64,11 +66,20 @@ async function activateLegacy(
     serviceContainer: IServiceContainer
 ) {
     // register "services"
-    const jupyterOutputChannel = window.createOutputChannel(localize.OutputChannelNames.jupyter());
-    const standardOutputChannel = jupyterOutputChannel;
+    const isDevMode =
+        !isTestExecution() &&
+        (context.extensionMode === ExtensionMode.Development ||
+            workspace.getConfiguration('jupyter').get<boolean>('development', false));
+    serviceManager.addSingletonInstance<boolean>(IsDevMode, isDevMode);
+
+    const standardOutputChannel = window.createOutputChannel(localize.OutputChannelNames.jupyter());
     addOutputChannelLogging(standardOutputChannel);
     serviceManager.addSingletonInstance<OutputChannel>(IOutputChannel, standardOutputChannel, STANDARD_OUTPUT_CHANNEL);
-    serviceManager.addSingletonInstance<OutputChannel>(IOutputChannel, jupyterOutputChannel, JUPYTER_OUTPUT_CHANNEL);
+    serviceManager.addSingletonInstance<OutputChannel>(
+        IOutputChannel,
+        getJupyterOutputChannel(isDevMode, standardOutputChannel),
+        JUPYTER_OUTPUT_CHANNEL
```

Comment:
This is what I discussed in a meeting/
* In dev mode (either when debugging or if flag is set)
* Then we have a custom output panel to see output from the Jupyter server.
* I have been doing this for a while now, 
* I do the same for kernel processes as well (in kernelProcess.ts i create new output chanels so i can see the output for each kernel, easier when debugging and working on issue)
	* However I'm not merging that change